bot/exts/moderation/infraction/_utils.py

```diff
@@ -98,7 +98,7 @@ async def post_infraction(
     if any(
         is_in_category(ctx.channel, category)
         for category in (Categories.modmail, Categories.appeals, Categories.appeals_2)
-    ):
+    ) or is_voice_channel(ctx.channel):
```

Comment:
Wouldn't it make more sense to check if ctx.message is not None?

Reply:
Yes it would, and that's the first obvious solution.

I just got lost in a tunnel vision of figuring out where it started from and following through.

I'll change is to that, as it's a much much simpler fix.

Reply:
I think we should consider (probably not in this PR) changing the type hints for any functions that can take a `FakeContext`, since just `Context` is misleading and makes it easy to introduce bugs like this.

Reply:
> I think we should consider (probably not in this PR) changing the type hints for any functions that can take a `FakeContext`, since just `Context` is misleading and makes it easy to introduce bugs like this.

I think the issue now is that we have 3 context classes: `Context`, `FakeContext` and `FilterContext` that are being used for conversion, and `post_infraction` takes either `Context` or `FakeContext`.

In this particular case, I think it's an instance of `FilterContext` that's being converted to `FakeContext` at the level of the `InfractionAndNotification.action` method .

But I do agree that it might need to be cleaned up.

I'll try to tackle this next.

Reply:
I'm open to a solution that gets rid of FakeContext, but as far as I could see it's a necessity since Context is created from either a message or an interaction.